File: client/src/app/models/Divida.ts
import DividaParcela from './DividaParcela';

export default class Divida {
  id: number;
  numeroTitulo: string;
  nomeDevedor: string;
  cpfDevedor: string;
  porcentagemJuros: number;
  porcentagemMulta: number;
  parcelas: DividaParcela[] = [];

  updateFields(newValues: Divida) {
    const {
      id,
      numeroTitulo,
      nomeDevedor,
      cpfDevedor,
      porcentagemJuros,
      porcentagemMulta,
    } = newValues;

    this.id = id || 0;
    this.numeroTitulo = numeroTitulo;
    this.nomeDevedor = nomeDevedor;
    this.cpfDevedor = cpfDevedor;
    this.porcentagemJuros = porcentagemJuros;
    this.porcentagemMulta = porcentagemMulta;
  }
}

File: Models/Divida.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace DesafioFULL.Models
{
  public class Divida
  {
    [Key]
    [Display(Name = "Id")]
    public int Id { get; set; }
    public int NumeroTitulo { get; set; }
    public string NomeDevedor { get; set; }
    public string CPFDevedor { get; set; }
    public int PorcentagemMulta { get; set; }
    public int PorcentagemJuros { get; set; }
    public List<DividaParcela> Parcelas { get; set; } = new List<DividaParcela>();
  }
}